### ISSUE TITLE
How can I get the result of a single image?

### ISSUE THREAD

**Issue by 2793145003** [decurtoydiaz/learning_with_signatures#5]
Hi, I'm trying to run your code and test the trained model on my own image.
After getting `Accuracy: 1.0` on all validation set, I upload an image and try to get it's classification.
Here is my code:
```
for c2 in range(0,categories):
  folder = '/content/'
  img = 'test.jpg'
  rmse_c = np.empty(categories, dtype='object')
  for c in range(0,categories):
    rmse_c[c] = mean_squared_error(globals()['supermeanl_' + str(c2)] * supermeanA[c], signature_cyz(folder, img), squared=False)
  min_rmse = np.argmin(rmse_c)
  print(labels[min_rmse]) # Can I get result like this?
```
And I get result:
```
cat
dog
wild
```
I'm not sure I write the code correctly, and I don't know how to improve it.
Please help, thanks!

**Comment by decurtoydiaz** [comment]
> Hi, I'm trying to run your code and test the trained model on my own image. After getting `Accuracy: 1.0` on all validation set, I upload an image and try to get it's classification. Here is my code:
> 
> ```
> for c2 in range(0,categories):
>   folder = '/content/'
>   img = 'test.jpg'
>   rmse_c = np.empty(categories, dtype='object')
>   for c in range(0,categories):
>     rmse_c[c] = mean_squared_error(globals()['supermeanl_' + str(c2)] * supermeanA[c], signature_cyz(folder, img), squared=False)
>   min_rmse = np.argmin(rmse_c)
>   print(labels[min_rmse]) # Can I get result like this?
> ```
> 
> And I get result:
> 
> ```
> cat
> dog
> wild
> ```
> 
> I'm not sure I write the code correctly, and I don't know how to improve it. Please help, thanks!

Thanks for your interest. Code is under development. Please wait until acceptance. Paper is only a draft. Will keep updating the repository.

De Curtò y DíAz.

**Comment by bicsi** [comment]
@2793145003 also see  #1 #2 #3 #4

**Comment by TechnikEmpire** [comment]
@decurtoydiaz Just wondering why the results were pulled from papers with code?

I am not meaning to be derisive or negative as others have been in comments, I'm just genuinely curious. Thanks.

**Comment by decurtoydiaz** [comment]
> @decurtoydiaz Just wondering why the results were pulled from papers with code?
> 
> I am not meaning to be derisive or negative as others have been in comments, I'm just genuinely curious. Thanks.

Thanks guys for your interest. 
Regarding papers with code. Somebody erase them. Not me. Anyway, as I said. This is only a draft. And what the current notebook shows is that if we are able to select among the correct lambdas, we have a set of optimal solutions that generalize very well. Those lambdas come from a geometric constraint, for example a set of 9 equations in AFHQ, so we should be able to determine the correct one to use at test time. Being by optimization, geometrical intuition, or some other way (several methods could be useful for this). 

Code is under development. Please wait until acceptance. Paper is only a draft. Will keep updating the repository.

De Curtò y DíAz.

**Comment by TechnikEmpire** [comment]
> if we are able to select among the correct lambdas, we have a set of optimal solutions that generalize very well. Those lambdas come from a geometric constraint, for example a set of 9 equations in AFHQ, so we should be able to determine the correct one to use at test time. Being by optimization, geometrical intuition, or some other way (several methods could be useful for this).

Can you clarify? Your remarks here seem to imply that this aspect is unsolved or undemonstrated in the current code. 

Anyway, I'm just going to try my own implementation since all the criticism given on tickets here are with alleged bugs in your code. However it seems lots of people don't understand the concept of a one-vs-all classifier. One objection with the initial iteration of the code (where I assume OVA was not fully implemented or explained) was that you were directly referencing the expected label in inference, which of course you would naturally do per-class in OVA.

I've done nothing more than skim over your stuff and comments thus far so thanks for your patience with the questions.

**Comment by 781458112** [comment]
@2793145003 I wrote the same code as you, but I found that no matter what I typed in, the output would contain all the categories of tags, whereas in fact I only found dozens of images from a single category, so as far as I am concerned, there is a problem with the last inference part of the code, right

**Comment by decurtoydiaz** [comment]
Code is correct. We only compute signatures and compare its spectrum. You can re-implement it if you want but the only thing that is not yet explained is how to choose the correct lambda at test time. 

De Curtò y DíAz.

**Comment by 2793145003** [comment]
> @2793145003 I wrote the same code as you, but I found that no matter what I typed in, the output would contain all the categories of tags, whereas in fact I only found dozens of images from a single category, so as far as I am concerned, there is a problem with the last inference part of the code, right

I also picked a image from training set, and the result is cat/dog/wild too. 
So I'm wondering how to choose a correct lambda. It's too weird for choosing lambda for every image at inference( if I do that, why don't I just get the images labeled), so maybe ... by using another model...?

**Comment by decurtoydiaz** [comment]
Yep, right. That is not yet explained.

De Curtò y DíAz.
